### PR TITLE
[connectivity] Bring "experimental" connectivity web to main repo.

### PR DESCRIPTION
**Pull request by ditman**
## Description

Connectivity for web has been baking for a while as "experimental_connectivity_web"; and as far as I know, we haven't received any major concerns with it.

This PR brings "experimental_connectivity_web" up to date with the latest master, and renames it to "connectivity_for_web" so it can become an endorsed implementation of "connectivity".

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/connectivity/connectivity/example/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+generated_plugin_registrant.dart
```
Comment (ditman) [inline packages/connectivity/connectivity/example/.gitignore:1]: Reset all the changes to connectivity/connectivity. They need to be brought back post-federation.

Comment (ditman) [reply packages/connectivity/connectivity/example/.gitignore]: Done.

File: packages/connectivity/connectivity_for_web/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+Find the example wiring in the [Google sign-in example application](https://github.com/ditman/plugins/blob/connectivity-web/packages/connectivity/connectivity/example/lib/main.dart).
+
+## Limitations on the web platform
+
+In order to retrieve information about the quality/speed of a browser's connection, the web implementation of the `connectivity` plugin uses the browser's [**NetworkInformation** Web API](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation), which as of this writing (June 2020) is still "experimental", and not available in all browsers:
+
+![Data on support for the netinfo feature across the major browsers from caniuse.com](https://caniuse.bitsofco.de/image/netinfo.png)
+
+On desktop browsers, this API only returns a very broad set of connectivity statuses (One of `'slow-2g', '2g', '3g', or '4g'`), and may *not* provide an Stream of changes. Firefox still hasn't enabled this feature by default.
```
Comment (yjbanov) [inline packages/connectivity/connectivity_for_web/README.md:31]: nit: a Stream

File: packages/connectivity/connectivity_for_web/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+![Data on support for the online-status feature across the major browsers from caniuse.com](https://caniuse.bitsofco.de/image/online-status.png)
+
+
+The NavigatorOnLine API is [provided by `dart:html`](https://api.dart.dev/stable/2.7.2/dart-html/Navigator/onLine.html), and only supports a boolean connectivity status (either online or offline), with no network speed information. In those cases the plugin will return either `wifi` (when the browser is online) or `none` (when it's not).
+
+Other than the approximate "downlink" speed, where available, and due to security and privacy concerns, **no Web browser will provide** any specific information about the actual network your users' device is connected to, like **the SSID on a Wi-Fi, or the MAC address of their device.**
+
+## Contributions and Testing
+
+Tests are a crucial to contributions to this package. All new contributions should be reasonably tested.
```
Comment (yjbanov) [inline packages/connectivity/connectivity_for_web/README.md:46]: nit: are crucial

File: packages/connectivity/connectivity_for_web/README.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+In order to run tests in this package, do:
+
+```
+cd test
+flutter run -d chrome
```
Comment (yjbanov) [inline packages/connectivity/connectivity_for_web/README.md:52]: `flutter test`?

File: packages/connectivity/connectivity_for_web/lib/connectivity_for_web.dart [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+      message:
+          'requestLocationServiceAuthorization() is not supported on the web platform.',
+    );
+  }
+
+  /// Get the current location service authorization (Only on iOS).
+  @override
+  Future<LocationAuthorizationStatus> getLocationServiceAuthorization() {
+    throw PlatformException(
+      code: 'UNSUPPORTED_OPERATION',
```
Comment (yjbanov) [inline packages/connectivity/connectivity_for_web/lib/connectivity_for_web.dart:66]: To reduce code size can this be moved into a shared function? Then you can just do `throw _unsupported('getLocationServiceAuthorization()';`

File: packages/connectivity/connectivity_for_web/lib/src/dart_html_connectivity_plugin.dart [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+import 'package:connectivity_platform_interface/connectivity_platform_interface.dart';
+import 'package:connectivity_for_web/connectivity_for_web.dart';
+
+/// The web implementation of the ConnectivityPlatform of the Connectivity plugin.
+class DartHtmlConnectivityPlugin extends ConnectivityPlugin {
+  /// Checks the connection status of the device.
+  @override
+  Future<ConnectivityResult> checkConnectivity() async {
+    return html.window.navigator.onLine
```
Comment (yjbanov) [inline packages/connectivity/connectivity_for_web/lib/src/dart_html_connectivity_plugin.dart:12]: Heads up: these things will become non-null soon. Consider performing null checks using JS-interop, or this hack:

```dart
bool _unsafeIsNull(dynamic object) => object == null;
```

(I'm not sure if this hack will work for long though; dart2js can inline it and decide to elide the null check)

File: packages/connectivity/connectivity_for_web/lib/src/utils/connectivity_result.dart [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  return ConnectivityResult.none;
+}
+
+ConnectivityResult _effectiveTypeToConnectivityResult(String effectiveType) {
+  // Possible values:
+  /*'2g'|'3g'|'4g'|'slow-2g'*/
+  switch (effectiveType) {
+    case 'slow-2g':
+    case '2g':
+    case '3g':
```
Comment (yjbanov) [inline packages/connectivity/connectivity_for_web/lib/src/utils/connectivity_result.dart:34]: What about 4g?

File: packages/connectivity/connectivity_for_web/README.md (hunk 4 above)
Comment (ditman) [reply packages/connectivity/connectivity_for_web/README.md:52]: Not yet. This requires the browser, so it's an unfinished e2e/driver test.

File: packages/connectivity/connectivity_for_web/README.md (hunk 2 above)
Comment (ditman) [reply packages/connectivity/connectivity_for_web/README.md:31]: Good catch!

File: packages/connectivity/connectivity_for_web/README.md (hunk 3 above)
Comment (ditman) [reply packages/connectivity/connectivity_for_web/README.md:46]: Thanks!

File: packages/connectivity/connectivity_for_web/lib/connectivity_for_web.dart (hunk 5 above)
Comment (ditman) [reply packages/connectivity/connectivity_for_web/lib/connectivity_for_web.dart:66]: Yep, I've done this.

File: packages/connectivity/connectivity_for_web/lib/src/dart_html_connectivity_plugin.dart (hunk 6 above)
Comment (ditman) [reply packages/connectivity/connectivity_for_web/lib/src/dart_html_connectivity_plugin.dart:12]: Hm `onLine` is a boolean; I hope the whole chain is not-null and defaults to "false" when null safety comes!

File: packages/connectivity/connectivity_for_web/lib/src/utils/connectivity_result.dart (hunk 7 above)
Comment (ditman) [reply packages/connectivity/connectivity_for_web/lib/src/utils/connectivity_result.dart:34]: '4g' according to MDN is a fabled connection with 0 ping and infinite bandwidth. It falls through to the 'wifi' default.

**Pull request merged**